lib/tokenizer/basic_tokenizer.rb

```diff
@@ -17,15 +17,10 @@ def size(text)
         end
 
         def truncate(text, max_length)
-          # Fast track the common case where the text is already short enough.
-          return text if text.size < max_length
```

Comment:
maybe we can still fast track for  `text.size < max_length / 2` , I think it looks pretty safe, looking at examples?

Reply:
gives some could be 3 tokens long or maybe for ... we can shortcut on / 3 extremely safely for now. I worry about all the extra work for ultra short strings.

Reply:
> gives some could be 3 tokens long or maybe for ... we can shortcut on / 3 extremely safely for now. I worry about all the extra work for ultra short strings.

I'm concerned that there might be some special cases causing exceptions, considering that the token count calculation is not simply based on character length. Perhaps we could use environment variables to configure whether this acceleration is necessary?

Reply:
I follow, maybe we just merge this for now and see

Reply:
@wozulong `DISCOURSE_AI_STRICT_TOKEN_COUNTING: true` will the ENV to disable acceleration.